src/redux/auth-reducer.tsx

```diff
@@ -3,29 +3,66 @@ import { authAPI } from "../api/authAPI";
 const SET_USER_DATA = "SET_USER_DATA";
 const REGISTRATION_SUCCESS = "REGISTRATION_SUCCESS";
 
-export const setAuthUserData = (id, email, login, isAuth) => {
+type payloadType = {
+  id: null | number;
+  email: null | string;
+  login: null | string;
+  isAuth: boolean;
+};
+
+type SetUserDataActionType = {
+  type: typeof SET_USER_DATA;
+  payload: payloadType;
+};
+
+export const setAuthUserData = (
+  id: null | number,
+  email: null | string,
+  login: null | string,
+  isAuth: boolean
+): SetUserDataActionType => {
   return {
     type: SET_USER_DATA,
     payload: { id, email, login, isAuth },
   };
 };
 
-export const registrationSuccess = (success) => {
+type registrationSuccessActionType = {
```

Comment:
`registrationSuccessActionType` -> `RegistrationSuccessAction` all types UpperCased, variables - lowerCased.
no need to add word 'type' to the name of each type, it is useless